src/Compilers/CSharp/Test/Emit2/Semantics/ExtensionTypeTests.cs

```diff
@@ -19447,5 +19767,1026 @@ public class Nested { }
+{
+    public int this[params int[] i] { get { System.Console.Write($"{i[0]} {i[1]} "); return 42; } }
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+        comp.VerifyDiagnostics();
+
+        // PROTOTYPE Execute when adding support for emitting non-static members
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<ElementAccessExpressionSyntax>(tree, "new C()[1, 2]");
+        Assert.Equal("System.Int32 E.this[params System.Int32[] i] { get; }", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+        Assert.Empty(model.GetMemberGroup(memberAccess)); // PROTOTYPE need to fix the semantic model
+    }
+
+    [ConditionalFact(typeof(CoreClrOnly))]
+    public void ExtensionIndexerAccess_OptionalParameter()
+    {
+        var source = """
+System.Console.Write(new C()[1]);
+
+class C { }
+
+implicit extension E for C
+{
+    public int this[int i, int j = 2] { get { System.Console.Write($"{i} {j} "); return 42; } }
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+        comp.VerifyDiagnostics();
+
+        // PROTOTYPE Execute when adding support for emitting non-static members
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<ElementAccessExpressionSyntax>(tree, "new C()[1]");
+        Assert.Equal("System.Int32 E.this[System.Int32 i, [System.Int32 j = 2]] { get; }", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+        Assert.Empty(model.GetMemberGroup(memberAccess)); // PROTOTYPE need to fix the semantic model
+    }
+
+    [Fact]
+    public void ExtensionIndexerAccess_ComImportReceiver()
+    {
+        string source = """
+using System;
+using System.Runtime.InteropServices;
+
+I i = new C();
+System.Console.Write(i[1]);
+
+[ComImport]
+[Guid("1FC6664D-C61E-4131-81CD-A3EE0DD6098F")]
+[CoClass(typeof(C))]
+interface I { }
+
+class C : I { }
+
+implicit extension E for I
+{
+    public int this[ref int i] => throw null;
+}
+""";
+        var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+        comp.VerifyEmitDiagnostics(
+            // (16,21): error CS0631: ref and out are not valid in this context
+            //     public int this[ref int i] => throw null;
+            Diagnostic(ErrorCode.ERR_IllegalRefParam, "ref").WithLocation(16, 21)
+            );
+
+        var tree = comp.SyntaxTrees.First();
+        var model = comp.GetSemanticModel(tree);
+        var memberAccess = GetSyntax<ElementAccessExpressionSyntax>(tree, "i[1]");
+        Assert.Equal("System.Int32 E.this[ref System.Int32 i] { get; }", model.GetSymbolInfo(memberAccess).Symbol.ToTestDisplayString());
+        Assert.Empty(model.GetMemberGroup(memberAccess).ToTestDisplayStrings()); // PROTOTYPE need to fix the semantic model
+    }
 }
```

Comment:
Do we have tests for dictionary initializers, like `new C() { [0] = "a", [1] = "b" }`?